lib/elixir/lib/system.ex

```diff
@@ -471,6 +474,38 @@ defmodule System do
     :erlang.halt(String.to_charlist(status))
   end
 
+  @doc """
+  Carefully stops the Erlang runtime system.
+
+  All applications are taken down smoothly, all code is unloaded, and all ports
+  are closed before the system terminates by calling `halt/1`.
+
+  `status` must be a non-negative integer value which is returned by the
+  runtime system to the operating system.
+
+  Note that on many platforms, only the status codes 0-255 are supported
+  by the operating system.
+
+  For more information, see [`:init.stop/1`](http://erlang.org/doc/man/init.html#stop-1).
+
+  ## Examples
+
+      System.stop(0)
+      System.stop(1)
+
+  """
+  @spec stop() :: no_return
+  @spec stop(non_neg_integer | binary) :: no_return
+  def stop(status \\ 0)
+
+  def stop(status) when is_integer(status) do
```

Comment:
I'm also wondering what is the official position about such duplicating checks. This may help to notice mistake earlir, but adds more work for CPU (of course this is not important in this specific case).
Is there some kind of official document like Elixir Philosophy, The Zen of Elixir, etc?